sotopia/generation_utils/generate.py

```diff
@@ -486,9 +486,9 @@ async def agenerate(
         input_values["format_instructions"] = output_parser.get_format_instructions()
 
     if structured_output:
-        assert (
-            model_name == "gpt-4o-2024-08-06"
-        ), "Structured output is only supported in gpt-4o-2024-08-06"
+        assert model_name == "gpt-4o-2024-08-06" or model_name.startswith(
+            "custom"
+        ), "Structured output is only supported in gpt-4o-2024-08-06 or custom models"
```

Comment:
do all the custom models support structured output?

Reply:
No but this if branch is only triggered if they explicitly opt in for structured output. There is no way for us to verify whether an arbitrary custom model can use structured output.